apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/Session.cs

```diff
@@ -0,0 +1,175 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+
+namespace Google.Cloud.Spanner.V1
+{
+    public partial class Session : IDisposable
+    {
+        private readonly CancellationTokenSource _cancellation = new CancellationTokenSource();
+        private Task _keepAlive;
```

Comment:
Ah - hadn't noticed this was a partial class mixed with the proto.

I think it would be better to separate the proto from the other behaviour, to be honest. Keeping all protos as pure DTOs makes them easy to reason about. We can't easily "hide" this class, but we could have a separate SpannerSession class perhaps?